classes/ability.php

```diff
@@ -481,7 +481,8 @@ function (submission $submission) {
                         $this->set_message('Cannot submit past the deadline');
                         return true;
                     } else {
-                        if ($submission->persisted()) {
+                        if ($submission->persisted() &&
+                            $submission->finalisedstatus != submission::FINALISED_STATUS_MANUALLY_UNFINALISED) {
```

Comment:
I much prefer indentation as follows:
```
if (
    $submission->persisted()
    &&
    $submission->finalisedstatus != submission::FINALISED_STATUS_MANUALLY_UNFINALISED
) {
```

In my view it's substantially clearer.